contracts/modules/VersionManager.sol

```diff
@@ -169,7 +169,12 @@ contract VersionManager is IVersionManager, IModule, BaseFeature, Owned {
      */
     function isFeatureAuthorised(address _wallet, address _feature) public view override returns (bool) {
         uint256 version = walletVersions[_wallet];
-        return isFeatureInVersion[_feature][version] || _feature == address(this) || (version == 0 && _feature == walletFactory);
+        return isFeatureInVersion[_feature][version] ||
+            // The VersionManager is the only feature that isn't stored in isFeatureInVersion
+            _feature == address(this) ||
+            // Temporary give the factory or an upgrader module the same permissions as features
+            // when a wallet is not yet initialized (version == 0)
+            (version == 0 && (_feature == walletFactory || IWallet(_wallet).authorised(_feature)));
```

Comment:
I'm not sure I like this as we are adding exceptions to  `isFeatureAuthorised` which is used in many places while they are only needed when `upgradeWallet` is called. It makes reasoning about security more complicated.

I think it would be cleaner to directly add these conditions to `upgradeWallet`.

```
function isFeatureAuthorised(address _wallet, address _feature) public view override returns (bool) {
     return isFeatureInVersion[_feature][version] || _feature == address(this);
}

function upgradeWallet(address _wallet, uint256 _toVersion) external override onlyWhenUnlocked(_wallet) {
        uint256 fromVersion = walletVersions[_wallet];
        require(isFeatureAuthorised(_wallet, msg.sender) || fromVersion == 0 && ((_feature == walletFactory || IWallet(_wallet).authorised(msg.sender)), "VM: caller cannot call upgrade");
        require(_toVersion > 0 && _toVersion <= lastVersion, "VM: Invalid _toVersion");
        require(fromVersion < _toVersion, "VM: Already on new version");
        walletVersions[_wallet] = _toVersion;
```